src/resources/auth/auth.controller.ts

```diff
@@ -46,3 +53,85 @@ export const signin: RequestHandler = (req, res, next) => {
 		}
 	})(req, res, next)
 }
+
+/**
+ * Forget password
+ * Save a reset password token and reset password expire to user model
+ * Send user a link that has the reset password token
+ *
+ */
+export const forgetPassword: RequestHandler = async (req, res, next) => {
+	// Check if email that user submitted belongs to an user
+	const {email} = req.body
+	try {
+		let user = await findUserWithEmail(email)
+		if (!user) {
+			const message = 'Could not find an user with provided email'
+			next(apiError.notFound(message))
+			return
+		}
+		// Create reset password token
+		const resetPasswordToken = uuidv4()
+		// Set expired time to be 1 hour
+		const resetPasswordExp = Date.now() + 3600000
+		// Save them to user object
+		user.resetPasswordToken = resetPasswordToken
+		user.resetPasswordExp = resetPasswordExp
+		// Save user to the database
+		await saveUser(user)
+		// Send an email to user, containing the reset password token
+		const resetUrl = `${req.headers.host}/auth/password/reset/${
+			user.resetPasswordToken
+		}`
+
+		const message = {
+			from: process.env.MAIL_SENDER,
```

Comment:
MAIL_SENDER should go to prod config file and default options for dev and test config 

- Dev <noreply@dev.com>
- Test <noreply@test.com>